src/controller/product-category-controller.ts

```diff
@@ -74,6 +74,8 @@ export default class ProductCategoryController extends BaseController {
    * @operationId getAllProductCategories
    * @tags productCategories - Operations of productcategory controller
    * @security JWT
+   * @param {boolean} onlyRoot.query - Whether to return only root categories
+   * @param {boolean} onlyLeaf.query - Whether to return only leaf categories
    * @param {integer} take.query - How many product categories the endpoint should return
    * @param {integer} skip.query - How many product categories should be skipped (for pagination)
    * @return {PaginatedProductCategoryResponse} 200 - All existing productcategories
```

Comment:
NIT: Could you remove the pagination from the response? In almost all cases you want all categories, and this just complicates front end code.

Reply:
This will make this PR a breaking change, if that is what you want?